addons/account/tests/test_account_move_entry.py

```diff
@@ -516,28 +519,120 @@ def test_entry_reverse_storno(self):
+        self.assertRecordValues(reversed_move_2.line_ids, [
             {
                 **self.entry_line_vals_1,
-                'debit': -500.0,
+                'debit': 500.0,
                 'credit': 0.0,
             }, {
                 **self.entry_line_vals_2,
                 'debit': 0.0,
-                'credit': -500.0,
-            }
+                'credit': 500.0,
+            },
         ])
+        self.assertEqual(reversed_move_2.amount_total_signed, 500.0)
+        self.assertEqual(reversed_move_2.amount_total_in_currency_signed, 500.0)
+
+    def test_manual_entries_storno(self):
+        # Test creating manual entries with negative amounts
+        # while Storno disabled/enabled and reversing them
+
+        line_1 = {
+            'name': 'Line 1',
+            'account_id': self.company_data['default_account_revenue'].id,
+            'debit': 0.0,
+            'credit': -1000.0,
+        }
+        line_2 = {
+            'name': 'Line 2',
+            'account_id': self.company_data['default_account_expense'].id,
+            'debit': -1000.0,
+            'credit': 0.0,
+        }
+
+        def manual_entry(is_storno, normal_assert_vals, reverse_assert_vals):
+            """Helper function to create a manual entry and check its reversal."""
+            self.env.company.account_storno = is_storno
+            move = self.env['account.move'].create({
+                'move_type': 'entry',
+                'partner_id': self.partner_a.id,
+                'date': fields.Date.from_string('2025-01-01'),
+                'currency_id': self.other_currency.id,
+                'line_ids': [
+                    Command.create({**line_1, 'expected_amount': -1000.0}),
+                    Command.create({**line_2, 'expected_amount': -1000.0}),
+                ],
+            })
+            move.action_post()
+            self.assertRecordValues(move.line_ids, normal_assert_vals)
+            self.assertEqual(move.amount_total, -1000.0 if is_storno else 1000.0)
+            self.assertEqual(move.amount_total_signed, -1000.0 if is_storno else 1000.0)
+            self.assertEqual(move.amount_total_in_currency_signed, -1000.0 if is_storno else 1000.0)
+
+            move_reversal = self.env['account.move.reversal'].with_context(active_model="account.move", active_ids=move.ids).create({
+                'date': fields.Date.from_string('2025-01-01'),
+                'journal_id': move.journal_id.id,
+            })
+            reversal = move_reversal.refund_moves()
+            reversed_move = self.env['account.move'].browse(reversal['res_id'])
+            self.assertRecordValues(reversed_move.line_ids, reverse_assert_vals)
+            self.assertEqual(reversed_move.amount_total, 1000.0)
+            self.assertEqual(reversed_move.amount_total_signed, 1000.0)
+            self.assertEqual(reversed_move.amount_total_in_currency_signed, 1000.0)
+
+        # While Storno disabled
+        manual_entry(False,
```

Comment:
```suggestion
        manual_entry(
            is_storno=False,
```
Let's name all the arguments :wink: